src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -388,6 +407,31 @@ private JsonRpcRequest ReadRequest(JToken json)
                 args is JArray argsArray ? (object)PartiallyParsePositionalArguments(argsArray) :
                 null;
 
+            // If method is $/progress, get the progress instance from the dictionary and call Report
+            string method = json.Value<string>("method");
+
+            if (method.Equals("$/progress"))
```

Comment:
>method [](start = 16, length = 6)

avoid using the instance string.Equals method with a value you obtained from some other place. If `method==null` this would throw, when you'd probably want it to just skip. Instead, use the static method: `string.Equals(method, "$/progress")`